R/calculate_stay.R

```diff
@@ -0,0 +1,36 @@
+#' Calculate total length of stay
+#'
+#' @description Calculate the total length of stay between start_date and end_date.
+#' If the end_date is missing then use the dummy discharge date.
+#'
+#' @param data Data to calculate the total length of stay
+#' @param year The financial year in '1920' format
+#' @param start_date The admission/start date variable. e.g. record_keydate1
+#' @param end_date The discharge/end date variable. e.g record_keydate2
+#'
+#' @return a [tibble][tibble::tibble-package] with additional variable `stay`.
+#' If there is no end date use dummy discharge to calculate the total length of stay.
+#' @export
+#'
+#' @family date functions
+calculate_stay <- function(data, year, start_date, end_date) {
```

Comment:
Sorry if I misunderstand James yesterday. Are we going to add a parameter called quarter or something? If there was such a para input, then the end_date would be the date of the end of the quarter plus 1.